libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1161,25 +1206,53 @@ void WaveClip::ShiftBy(double delta) noexcept
 
 // Bug 2288 allowed overlapping clips.
 // This was a classic fencepost error.
-// We are within the clip if start < t <= end.
+// We are within the clip if start <= t < end.
 // Note that BeforeClip and AfterClip must be consistent
 // with this definition.
 bool WaveClip::WithinPlayRegion(double t) const
 {
-    auto ts = TimeToSamples(t);
-    return ts > GetPlayStartSample() && ts < GetPlayEndSample() + GetAppendBufferLen();
+   return GetPlayStartTime() <= t && t < GetPlayEndTime();
+}
+
+bool WaveClip::EntirelyWithinPlayRegion(double t0, double t1) const
+{
+   assert(t0 <= t1);
+   return !BeforePlayRegion(t0) && !AfterPlayRegion(t1);
+}
+
+bool WaveClip::PartlyWithinPlayRegion(double t0, double t1) const
```

Comment:
This function is true in case the play region is a proper subset of [t0, t1)

Reply:
Maybe it is correctly used but maybe the name misleads.

Reply:
> This function is true in case the play region is a proper subset of [t0, t1)

Are you sure? For a clip with boundaries `[0, 1)`, `PartlyWithinPlayRegion(0.2, 0.3)` would return false, wouldn't it?